telemetry/definitions/commonDefinitions.json

```diff
@@ -159,6 +159,17 @@
             "type": "string",
             "allowedValues": ["IAM", "SecretsManager"],
             "description": "How the database credentials are being retrieved"
+        },
+        {
+            "name": "snippetPrefix",
+            "type": "string",
+            "description": "The prefix of a snippet"
+        },
+        {
+            "name": "snippetLanguage",
```

Comment:
Can we use runtimeString instead of making more languages? (runtimeString can be renamed if you like because it's only used in a thing that is unused currently)

Reply:
That wouldn't tell us TS vs js.

Reply:
runtimeString is a generic runtime name that is currently only used with `aws_injectCredentials` (which is in draft PR in JB)

Reply:
I'm thinking it's not really a runtime. It corresponds to the language of the code (`javascript` vs. `nodejs`, `java` vs `java8` or `OpenJDK8`, `csharp` vs `dotnet`, etc.).

The exact correspondence I'm modelling is called [`language identifier`](https://code.visualstudio.com/docs/languages/identifiers#_known-language-identifiers) in vscode. Jetbrains calls these "contexts" under which the live templates are made available.

Also it seems useful to make an enum here so that we don't end up with `java` and `Java`.